pkg/config/config.go

```diff
@@ -446,14 +448,24 @@ func Load() error {
 		}
 	}
 
-	loadProxyFromEnv()
-	sanitizeAPIKey()
+	loadProxyFromEnv(Datadog)
+	sanitizeAPIKey(Datadog)
 	return nil
 }
 
 // Avoid log ingestion breaking because of a newline in the API key
-func sanitizeAPIKey() {
-	Datadog.Set("api_key", strings.TrimSpace(Datadog.GetString("api_key")))
+func sanitizeAPIKey(config Config) {
+	config.Set("api_key", strings.TrimSpace(config.GetString("api_key")))
+}
+
+// GetMainInfraEndpoint returns the main DD Infra URL defined in the config, based on the value of `site` and `dd_url`
+func GetMainInfraEndpoint() string {
+	return getMainInfraEndpoint(Datadog)
+}
+
+// GetMainEndpoint returns the main DD URL defined in the config, based on `site` and the prefix, or ddURLKey
```

Comment:
Where is this used ? I don't see it.

Reply:
it's used by process agent (see https://github.com/DataDog/datadog-process-agent/pull/192), could also be used by logs for example. Do you think it's worth making it more clear in the comment?

Reply:
:+1: